### PR TITLE
GH-126212: Fix removal of slashes in file URIs on Windows

### PR DESCRIPTION
**Pull request by barneygale**
Adjust `urllib.request.pathname2url()` and `url2pathname()` so that they don't remove slashes from Windows DOS drive paths and URLs. There was no basis for this behaviour, and it conflicts with how UNC and POSIX paths are handled.

* Issue: gh-126212

### CODE REVIEW

File: Lib/nturl2path.py [hunk 1]
```diff
@@ -24,22 +24,14 @@ def url2pathname(url):
             # convert this to \\host\path\on\remote\host
             # (notice halving of slashes at the start of the path)
             url = url[2:]
-        components = url.split('/')
-        # make sure not to convert quoted slashes :-)
```
Comment (barneygale) [inline Lib/nturl2path.py:28]: This comment implies that quoted forward slashes shouldn't be converted, so a URL like `file:///c:/foo%2fbar` should result in `c:\foo/bar`. But I can't see any reason for this, given forward and backward slashes are treated identically by most Windows programs, even going back to the first versions of Windows. Python path operations almost always treat them as interchangeable, e.g. plenty of code does `p.replace(os.path.altsep, os.path.sep)`

Comment (zooba) [reply Lib/nturl2path.py:28]: Unfortunately, we need to consider how this affects people who parse/convert and then do comparisons. There are a bunch of "security" researchers going around finding ways for blocklists to break[^1] and forcing us to announce vulnerabilities.

[^1]: Spoiler: blocklists are always broken. Use a safelist.

Comment (barneygale) [reply Lib/nturl2path.py:28]: Fair point! I've reverted the change and added a test.

**Comment by barneygale** [comment]
Hey @zooba, hope I'm not being rude with the ping, just wondering if you intend to review again? If so I'm happy to wait! Thanks

**Comment by zooba** [comment]
It's not rude, but it's also not helpful :) Pings don't reach me any quicker than regular notifications.

This looks good to me. Thanks!

**Comment by barneygale** [comment]
Thanks for the review! I pinged because you approved a closely-related PR at the same time, and I thought there might have been a mix-up between the two as you were so quick to reply initially. My mistake - I'm sorry for the noise. And thanks again.

**Pull request merged**